lighthouse-cli/bin.ts

```diff
@@ -0,0 +1,251 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _SIGINT = 'SIGINT';
+const _ERROR_EXIT_CODE = 130;
+const _RUNTIME_ERROR_CODE = 1;
+
+const environment = require('../lighthouse-core/lib/environment.js');
+if (!environment.checkNodeCompatibility()) {
+  console.warn('Compatibility error', 'Lighthouse requires node 5+ or 4 with --harmony');
+  process.exit(_RUNTIME_ERROR_CODE);
+}
+
+import * as path from 'path';
+const yargs = require('yargs');
+import * as Printer from './printer';
+const lighthouse = require('../lighthouse-core');
+const assetSaver = require('../lighthouse-core/lib/asset-saver.js');
+const log = require('../lighthouse-core/lib/log');
+import {ChromeLauncher} from './chrome-launcher';
+import * as Commands from './commands/commands';
+
+const perfOnlyConfig = require('../lighthouse-core/config/perf.json');
+
+const cli = yargs
+  .help('help')
+  .version(() => require('../package').version)
+  .showHelpOnFail(false, 'Specify --help for available options')
+
+  .usage('$0 url')
+
+  // List of options
+  .group([
+    'verbose',
+    'quiet'
+  ], 'Logging:')
+  .describe({
+    verbose: 'Displays verbose logging',
+    quiet: 'Displays no progress or debug logs'
+  })
+
+  .group([
+    'mobile',
+    'save-assets',
+    'save-artifacts',
+    'list-all-audits',
+    'list-trace-categories',
+    'config-path',
+    'perf',
+    'disable-network-throttling',
+  ], 'Configuration:')
+  .describe({
+    'mobile': 'Emulates a Nexus 5X',
+    'save-assets': 'Save the trace contents & screenshots to disk',
+    'save-artifacts': 'Save all gathered artifacts to disk',
+    'list-all-audits': 'Prints a list of all available audits and exits',
+    'list-trace-categories': 'Prints a list of all required trace categories and exits',
+    'config-path': 'The path to the config JSON.',
+    'perf': 'Use a performance-test-only configuration',
+    'skip-autolaunch': 'Skip autolaunch of chrome when accessing port 9222 fails',
+    'select-chrome': 'Choose chrome location to use when multiple installations are found',
+    'disable-network-throttling': 'Disable network throttling when running audits.',
+  })
+
+  .group([
+    'output',
+    'output-path'
+  ], 'Output:')
+  .describe({
+    'output': 'Reporter for the results',
+    'output-path': `The file path to output the results
+Example: --output-path=./lighthouse-results.html`
+  })
+
+  // boolean values
+  .boolean([
+    'save-assets',
+    'save-artifacts',
+    'list-all-audits',
+    'list-trace-categories',
+    'mobile',
+    'perf',
+    'skip-autolaunch',
+    'select-chrome',
+    'verbose',
+    'quiet',
+    'help'
+  ])
+  .choices('output', Printer.GetValidOutputOptions())
+
+  // default values
+  .default('mobile', true)
+  .default('output', Printer.GetValidOutputOptions()[Printer.OutputMode.pretty])
```

Comment:
we have a list of valid strings, and we have a named concepts for each of those strings.

This approach gives us the ability to decouple the string index + the representative named concept.

so given the constraints around this string "linked list" for string => index => string relationship I think yes